special-pages/pages/duckplayer/app/settings.js

```diff
@@ -7,20 +7,56 @@ export class Settings {
      * @param {{state: 'enabled' | 'disabled'}} [params.pip]
      * @param {{state: 'enabled' | 'disabled'}} [params.autoplay]
      * @param {{state: 'enabled' | 'disabled'}} [params.focusMode]
-     * @param {import("../types/duckplayer.js").InitialSetupResponse['settings']['customError']} [params.customError]
+     * @param {import("../types/duckplayer.js").DuckPlayerPageSettings['customError']} [params.customError]
      */
     constructor({
         platform = { name: 'macos' },
         pip = { state: 'disabled' },
         autoplay = { state: 'enabled' },
         focusMode = { state: 'enabled' },
-        customError = { state: 'disabled', signInRequiredSelector: '' },
+        customError = { state: 'disabled', settings: {}, signInRequiredSelector: '' },
     }) {
         this.platform = platform;
         this.pip = pip;
         this.autoplay = autoplay;
         this.focusMode = focusMode;
-        this.customError = customError;
+        this.customError = this.parseLegacyCustomError(customError);
+    }
+
+    /**
+     * Parses custom error settings so that both old and new schemas are accepted.
+     *
+     * Old schema:
+     * {
+     *   state: "enabled",
+     *   signInRequiredSelector: "div"
+     * }
+     *
+     * New schema:
+     * {
+     *   state: "disabled",
+     *   settings: {
+     *     signInRequiredSelector: "div"
+     *   }
+     * }
```

Comment:
sorry for not spotting this earlier - I do think it’s worth aligning with the subfeature shape in the config for now. Just to make future things easier.